### ISSUE TITLE
visibility: thread

### ISSUE THREAD

**Issue by Ovid** [mastodon/mastodon#26816]
### Pitch

(Apologies if this is already suggested. I can't find it)

When I reply to a post, it often shows up on the public timeline. If I set `visibility: unlisted`, it doesn't show up on the local timeline, but my followers can still see it. If I post a long thread, this spams their timeline.

I would love a `visibility: thread` option which hides a post. It will only be visible if:

1. It's not a reply
2. It's a reply and someone has clicked on a post in the thread

### Motivation

I recently posted a long thread with 12 posts. Someone complained that my 12 posts spammed their local timeline. They have every right to be unhappy about that.

So I read the docs and for every reply, I set `visibility: unlisted`. That solved the problem with the local timeline, but my followers apparently see the entire thread and that still spams their timeline (I unfollowed a very popular author because of their long threads).

A `visibility: thread` option would allow me to have a top-level post that might interest them, but not spam them with the replies unless they click through to the thread.

Similar issues:

* [Privacy Overhaul](https://github.com/mastodon/mastodon/issues/23551)
* ["Reply" post privacy level](https://github.com/mastodon/mastodon/issues/23406)
* [Default visibility when replying to post](https://github.com/mastodon/mastodon/issues/21745)
* [Option to not show "unlisted" replies in home timelines](https://github.com/mastodon/mastodon/issues/21245)

**Comment by dboehmer** [comment]
To be honest I thought `visibility: unlisted` meant that the toot will not be listed in the timeline of my followers. I support this proposal.

**Comment by derspyy** [comment]
#8615 related?

**Comment by pezcurrel** [comment]
I totally support this idea, and i think ti would be also useful if it was possible to have a visibility default of "first post public, following posts by the same author unlisted" and "first post public, following posts by the same author: thread" :)

**Comment by brendanjones** [comment]
Btw @Ovid may I suggest editing the issue name to something like “Add post visibility option: thread”? Might make it easier to find.

**Comment by Pxtl** [comment]
I like this too, but also for the reverse case.  Often I see high traffic influencers gripe about how they're explaining things over and over again in the replies, but nobody else sees those replies because they didnt get @ ed.  On the other site they'd quote-tweet, but that's not on Mastodon (yet?) and for good reason.  Be able to "loud reply" would be a good option, especially if the feed would realize that the context for that reply should be visible too.

**Comment by brendanjones** [comment]
@Pxtl I don’t quite follow. How would this loud reply work differently to a Public reply? Do you mean everyone in the thread gets notified of the comment even if they’re not mentioned?

**Comment by Pxtl** [comment]
> @Pxtl I don’t quite follow. How would this loud reply work differently to a Public reply? Do you mean everyone in the thread gets notified of the comment even if they’re not mentioned?

Public replies afaik do not appear on the feeds of your followers.  You have to manually retoot them.

Basically, Mastodon appears to have some hidden non-obvious visibility rule: your replies, except for self-replies, are not shown on your followers' feeds.

By reifying the reply visibility option but simply making it default for replies (and non-default for posts and self-replies) we allow users to override that behavior.

So when you're replying to a toot correcting somebody for being a dick, you can give it "public" visibility and make it visible to your followers (which is mechanically the same as retooting but self-retoot has egotist semantics people like to avoid) and then your followers can pile-on (it's vicious but public quote-tweet are how VIPs police their replies and push back against the disproportionate harassment they get on Twitter).

**Comment by brendanjones** [comment]
@Pxtl this is incorrect, public replies do appear in followers’ feeds.

**Comment by brendanjones** [comment]
@Pxtl I have no idea. Maybe check that you’re not filtering replies from your timeline? In any case, if you’d like a different post visibility option then would be good to make a separate issue for it and discuss there, to keep this issue on topic.

**Comment by Pxtl** [comment]
@brendanjones I have confirmed I am not filtering replies from my timeline.  I have confirmed I am not seeing non-self-replies from people I follow (maybe it depends on which instance they're on?).

Either way, I'm not asking for a new post visibility option.  The experience I'm describing *is* the "thread" visibility option as described by @Ovid above: these replies are not shown to your followers unless you investigate the thread or the poster.

So basically, Mastodon.social (the instance I use and the apparent flagship) currently *appears* to have an undocumented mode-switch:

1) Public posts and self-replies: Show them to all your followers.
2) Replies to others: do not show them to your followers. This is basically the "thread" visibility option described by @Ovid.

Also @Ovid is requesting is the ability to *manually* toggle from (1) to (2) when initiating a self-reply.  I'm pointing out that the reverse ability, which still uses the same 2 visibility modes, would also be useful (moving a (2) to a (1)) although that can be mimicked using a retoot.  That's why I believe that this is in-scope with the above feature.

If the fact that I do not see replies-to-others in my home-feed, despite having the "show replies" option enabled is a *bug*, then this is news to me.  I'd assumed this was intentional behavior and there was some stricter metric for when it will show "replies" in the filter (eg maybe it hides replies to people I don't follow, or it hides replies to other instances, or whatever).

tl;dr: based on the behavior I see in Mastodon.social, the requested "thread" visibility level **already exists**.  It is applied automatically in replies-to-others, regardless of filter modes.  So it only makes sense to allow users to be able to move their posts in *both directions*, from "thread" to "normal" and vice versa.  Hence, in-scope for this issue.

**Comment by brendanjones** [comment]
Ah apologies, I misunderstood what you were describing. 

I don’t know what to say about you not seeing replies, it seems odd that that’d be something only mastodon.social does. Are you sure you’re not just not noticing that they’re replies? In the default mastodon UI, there’s no ‘replying to’ above the post that tips you off like there is with self relies - the only way you can tell it’s a reply is that a username has been tagged in the post. 

This it’s not great design - apps I use like Mona make it very obvious when a post is a reply.

**Comment by Pxtl** [comment]
Just to be sure I wasn't crazy I checked my more prolific follows to be sure -- they're replying, and I'm not seeing it.

I think the behaviour is "you only see replies in cases where you follow both the sender *and* the recipient" to prevent the "half a conversation" problem that plagued early Twitter.

**Comment by brendanjones** [comment]
Ah, if that's the case then that'd be why I'm seeing replies when you're not; I follow 3x more people than you do, and plenty of them also follow each other.
